packages/web-frontend/src/utils/measures.js

```diff
@@ -349,14 +349,16 @@ export function getMeasureDisplayInfo(measureData, measureOptions, hiddenMeasure
         case MEASURE_TYPE_SHADED_SPECTRUM:
           displayInfo.originalValue =
             valueInfo.value === null || valueInfo.value === undefined ? 'No data' : valueInfo.value;
-          displayInfo.color = resolveSpectrumColour(
-            scaleType,
-            scaleColorScheme,
-            valueInfo.value || (valueInfo.value === 0 ? 0 : null),
-            min,
-            max,
-            noDataColour,
-          );
+          displayInfo.color =
+            displayInfo.color ||
```

Comment:
This kind of behaviour can be seen in this function already, however in general it seems fairly inconsistent.

It really is a bit of a hack to be honest and could definitely be talked into doing something more comprehensive...

Reply:
I got the same feeling as you did, but not sure if this change will effect some visuals? We previously always want to replace displayInfo.color with the last `measureOption`.
  
But I just don't have any guts to change it.

Reply:
I think we may be in the clear on this. I used this SQL query to check for the following: 
- Any shaded-spectrum mapOverlays with linked measures
- Any shaded-spectrum mapOverlays that were a linked measure of another overlay
```
select * from "mapOverlay" mo
where ("presentationOptions" ->>'displayType' = 'shaded-spectrum' and "linkedMeasures" != null)
or (id in (select unnest("linkedMeasures") as linked_measures
  from "mapOverlay" mo) and "presentationOptions" ->>'displayType' = 'shaded-spectrum')
```
I got 0 results, which I think means that 'for all map overlays with shaded-spectrum type, there is just a single measureOption for each'? So this change should be safe?

I feel we can still come up with a different approach to fixing this anyway, lets chat at some point today about it 👍